src/tests/ftest/util/ior_utils.py

```diff
@@ -1,110 +1,267 @@
+
+        """
+        self.value = test.params.get(name, path, self.default)
+
+
+class IorCommand():
+    """Defines a object representing a IOR command."""
+
+    def __init__(self):
+        """Create an IorCommand object."""
+        self.flags = IorParam("{}")                 # IOR flags
+        self.api = IorParam("-a {}", "DAOS")        # API for I/O
+        self.block_size = IorParam("-b {}")         # bytes to write per task
+        self.test_delay = IorParam("-d {}")         # sec delay between reps
+        self.script = IorParam("-f {}")             # test script name
+        self.signatute = IorParam("-G {}")          # set value for rand seed
+        self.repetitions = IorParam("-i {}")        # number of repetitions
+        self.outlier_threshold = IorParam("-j {}")  # warn if N sec from mean
+        self.alignment = IorParam("-J {}")          # HDF5 alignment in bytes
+        self.data_packet_type = IorParam("-l {}")   # type of packet created
+        self.memory_per_node = IorParam("-M {}")    # hog memory on the node
+        self.num_tasks = IorParam("-N {}")          # number of tasks
+        self.test_file = IorParam("-o {}")          # full name for test
+        self.directives = IorParam("-O {}")         # IOR directives
+        self.task_offset = IorParam("-Q {}")        # for -C & -Z read tests
+        self.segment_count = IorParam("-s {}")      # number of segments
+        self.transfer_size = IorParam("-t {}")      # bytes to transfer
+        self.max_duration = IorParam("-T {}")       # max mins executing test
+        self.daos_pool = IorParam("--daos.pool {}")             # pool uuid
+        self.daos_svcl = IorParam("--daos.svcl {}")             # pool SVCL
+        self.daos_cont = IorParam("--daos.cont {}")             # cont uuid
+        self.daos_destroy = IorParam("--daos.destroy", True)    # destroy cont
+        self.daos_group = IorParam("--daos.group {}")           # server group
+        self.daos_chunk = IorParam("--daos.chunk_size {}", 1048576)
+        self.daos_oclass = IorParam("--daos.oclass {}")         # object class
+
+    def __str__(self):
+        """Return a IorCommand object as a string.
+
+        Returns:
+            str: the ior command with all the defined parameters
+
+        """
+        params = []
+        for value in self.__dict__.values():
+            value_str = str(value)
+            if value_str != "":
+                params.append(value_str)
+        return " ".join(["ior"] + sorted(params))
+
+    def set_params(self, test, path="/run/ior/*"):
+        """Set values for all of the ior command params using a yaml file.
+
+        Args:
+            test (Test): avocado Test object
+            path (str, optional): yaml namespace. Defaults to "/run/ior/*".
+
+        """
+        for name, ior_param in self.__dict__.items():
+            ior_param.set_yaml_value(name, test, path)
+
+    def get_launch_command(self, basepath, processes, hostfile):
```

Comment:
To support different paths to the mpirun/oretrun command, we need an additional argument, something like:

`def get_launch_command(self, basepath, processes, hostfile, run_path=None):`